lib/webserver.js

```diff
@@ -40,6 +45,26 @@ function main(
     }),
   );
   app.use(stripEmptyRequestBody);
+  app.use((err, req, res, next) => {
+    const brokerToken = extractBrokerTokenFromUrl(req.url);
+    const logContext = {
+      url: request.url,
+      requestMethod: req.method,
+      requestHeaders: req.headers,
+      requestId: req.headers['snyk-request-id'] || '',
+      maskedToken: maskToken(brokerToken),
+    };
+
+    logger.error(
+      {
+        logContext,
+        error: err,
+        stackTrace: new Error('stacktrace generator').stack,
+      },
+      'Error occurred receiving http request on broker webserver',
+    );
+    next();
+  });
```

Comment:
I think this is too late to catch the error we had - looking at the stack trace I saw locally, it was

```
PayloadTooLargeError: request entity too large
 &nbsp; &nbsp;at readStream (/Users/athorburn/workspace/broker/node_modules/raw-body/index.js:163:17)
 &nbsp; &nbsp;at getRawBody (/Users/athorburn/workspace/broker/node_modules/raw-body/index.js:116:12)
 &nbsp; &nbsp;at read (/Users/athorburn/workspace/broker/node_modules/body-parser/lib/read.js:79:3)
 &nbsp; &nbsp;at rawParser (/Users/athorburn/workspace/broker/node_modules/body-parser/lib/types/raw.js:81:5)
 &nbsp; &nbsp;at Layer.handle [as handle_request] (/Users/athorburn/workspace/broker/node_modules/express/lib/router/layer.js:95:5)
 &nbsp; &nbsp;at trim_prefix (/Users/athorburn/workspace/broker/node_modules/express/lib/router/index.js:328:13)
 &nbsp; &nbsp;at /Users/athorburn/workspace/broker/node_modules/express/lib/router/index.js:286:9
 &nbsp; &nbsp;at Function.process_params (/Users/athorburn/workspace/broker/node_modules/express/lib/router/index.js:346:12)
 &nbsp; &nbsp;at next (/Users/athorburn/workspace/broker/node_modules/express/lib/router/index.js:280:10)
 &nbsp; &nbsp;at markEmptyRequestBody (/Users/athorburn/workspace/broker/dist/lib/webserver.js:13:5)
```

So I think this needs to move to _before_
```
app.use(
    bodyParser.raw({
      type: (req) =>
        req.headers['content-type'] !==
        'application/vnd.broker.stream+octet-stream',
      limit: '10mb',
     }),
   );
```

Reply:
moved